source/MaterialXCore/Types.cpp

```diff
@@ -185,10 +220,47 @@ template <> Matrix44 MatrixN<Matrix44, float, 4>::getAdjugate() const
         _arr[0][0]*_arr[2][1]*_arr[1][2] - _arr[1][0]*_arr[0][1]*_arr[2][2] - _arr[2][0]*_arr[1][1]*_arr[0][2]);
 }
 
+
+Vector4 Matrix44::transform(const Vector4& rhs) const
```

Comment:
For these pure matrix/vector multiplication methods, I would recommend the original name "multiply", with the name "transform" being reserved for the specific point/vector/normal cases.